neureka/hal/neureka.c

```diff
@@ -18,15 +18,22 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#ifndef __NE16_ERROR_CODES_H__
-#define __NE16_ERROR_CODES_H__
+#include "neureka.h"
 
-typedef enum {
-  success = 0,
-  weightBitwidthOutOfBounds,
-  unsupportedWeightOffsetMode,
-  unsupportedFeatureBitwidth,
-  dimensionMismatch
-} nnx_error_code;
+#define NEUREKA_STATUS_EMPTY (0x000)
+#define NEUREKA_STATUS_FULL (0x101)
 
-#endif // __NE16_ERROR_CODES_H__
\ No newline at end of file
+inline int neureka_task_queue_size(neureka_dev_t *dev) { return 2; }
```

Comment:
:)

Reply:
Yes?

Reply:
Is the task queue size something that derives from the neureka_dev_t struct? Is it something that might vary with future implementations? If not I think a `#define` might be a good solution worth considering.

Reply:
I believe this is the reflection of a hardware parameter that can be 2, 2, or sometimes 2. Just kidding. It can also be 1 (to define at accelerator design time) or, in theory, up to 4 with small changes, but in practice 2 works better than 1 at little added cost and it is very hard to find a use case for >2.
So I'd go for the `#define` :D

Reply:
Changed here https://github.com/pulp-platform/pulp-nnx/pull/2/commits/1a4f873327ea3b7271e0a313a34ce3661bad5d21